snippets/graph_neighbourhood.py

```diff
@@ -0,0 +1,68 @@
+""""""
+
+from itertools import starmap
+from operator import le, lt, and_
+from pathlib import Path
+from os import path
+from sys import argv, stdout
+from argparse import ArgumentParser
+
+from data import load_graph
+
+
+def neighbourhood(coord: tuple[int, int], borders) -> None:
+    n, s, w, e, nw, ne, sw, se = [
+        (-1, 0),
+        (1, 0),
+        (0, -1),
+        (0, 1),
+        (-1, -1),
+        (-1, 1),
+        (1, -1),
+        (1, 1),
+    ]
+    return filter(
+        lambda l: and_(*starmap(le, zip((0, 0), l)))
+        and and_(*starmap(lt, zip(l, borders))),
+        map(lambda n: tuple(map(sum, zip(coord, n))), (n, s, w, e, nw, ne, sw, se)),
+    )
+
+
+if __name__ == "__main__":
+    parser = ArgumentParser(
+        description="Determine the list of arcs in the input Graph whose weight is greater or equal to their neighbour coordinates."
```

Comment:
Could you clarify whether an 'arc' is another term for 'edge' or has some special technical meaning - ie is it a special type of 'edge'.